beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -242,6 +242,14 @@ func (vs *Server) getPayloadHeaderFromBuilder(ctx context.Context, slot primitiv
 		return nil, fmt.Errorf("incorrect parent hash %#x != %#x", header.ParentHash(), h.BlockHash())
 	}
 
+	reg, err := vs.BeaconDB.RegistrationByValidatorID(ctx, idx)
```

Comment:
we shouldn't get it from here, it should be from vs.BlockBuilder.RegistrationByValidatorID() instead, there is a cache here.